### PR TITLE
Improve error message for arg type unsupported during typing.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
This make the error message look like e.g.:
```python
$ ipython
Python 3.6.5 |Anaconda, Inc.| (default, Apr 29 2018, 16:14:56) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numpy as np

In [2]: from numba import njit

In [3]: njit(lambda : np.linalg.svd([1]))()
---------------------------------------------------------------------------
TypingError                               Traceback (most recent call last)
<ipython-input-3-0393378f98ef> in <module>()
----> 1 njit(lambda : np.linalg.svd([1]))()

<env>/numba/dispatcher.py in _compile_for_args(self, *args, **kws)
    347                 e.patch_message(msg)
    348 
--> 349             error_rewrite(e, 'typing')
    350         except errors.UnsupportedError as e:
    351             # Something unsupported is present in the user code, add help info

<env>/numba/dispatcher.py in error_rewrite(e, issue_type)
    314                 raise e
    315             else:
--> 316                 reraise(type(e), e, None)
    317 
    318         argtypes = []

<env>/numba/six.py in reraise(tp, value, tb)
    656             value = tp()
    657         if value.__traceback__ is not tb:
--> 658             raise value.with_traceback(tb)
    659         raise value
    660 

TypingError: Failed at nopython (nopython frontend)
Invalid use of Function(<function svd at 0x7f6a8092db70>) with argument(s) of type(s): (list(int64))
 * parameterized
In definition 0:
    TypingError: np.linalg.svd() only supported for array types
    raised from <path>/numba/numba/targets/linalg.py:739
This error is usually caused by passing an unsupported argument type to the named function.
[1] During: resolving callee type: Function(<function svd at 0x7f6a8092db70>)
[2] During: typing of call at <ipython-input-3-0393378f98ef> (1)


File "<ipython-input-3-0393378f98ef>", line 1:
njit(lambda : np.linalg.svd([1]))()
^

This is not usually a problem with Numba itself but instead often caused by
the use of unsupported features or an issue in resolving types.
```

File: numba/types/functions.py [hunk 1]
```diff
@@ -53,6 +53,9 @@ def format(self):
             if loc:
                 msgbuf.append('{}raised from {}'.format(indent, loc))
 
+        likely_cause = ("This error is usually caused by passing an "
+                        "unsupported argument type to the named function.")
+        msgbuf += [_termcolor.errmsg(likely_cause)]
```
Comment (hameerabbasi) [inline numba/types/functions.py]: If `msgbuf` is a `list`, couldn't you just do `msgbuf.append(_termcolor.errmsg(likely_cause))`? This syntax is a bit opaque IMO.

Comment (stuartarchibald) [reply numba/types/functions.py]: thanks. yes. changed.

File: numba/tests/test_ctypes.py [hunk 2]
```diff
@@ -217,7 +217,7 @@ def test_passing_array_ctypes_voidptr(self):
         self.check_array_ctypes(use_c_vsquare)
 
     @tag('important')
-    def test_passing_array_ctypes_voidptr(self):
+    def test_passing_array_ctypes_voidptr_pass_ptr(self):
```
Comment (stuartarchibald) [inline numba/tests/test_ctypes.py]: there were two tests called `test_passing_array_ctypes_voidptr`, only one ran as a result, hence rename.

**Comment by seibert** [comment]
The change in error message requires updating this test (now failing on Travis CI):

https://github.com/numba/numba/blob/b8151838cc195d0ea6974b618ec22eed02a56fa2/numba/tests/test_operators.py#L1039-L1052

**Comment by stuartarchibald** [comment]
@seibert thanks, fixed that one too :)

**Pull request merged**